fvm/handler/event.go

```diff
@@ -59,7 +59,8 @@ func (h *EventHandler) EmitEvent(event cadence.Event,
 	}
 
 	flowEvent := flow.Event{
-		Type:             flow.EventType(event.EventType.ID()),
+		// Type:             flow.EventType(event.EventType.ID()),
+		Type:             flow.EventType(event.EventType.QualifiedIdentifier), // XXX is this safe?
```

Comment:
Need to change this to include service account address, as before

Reply:
I am not sure about this.

The function that checks if the event is a service event actually uses another field (`location`) of the event to see if it was emitted by the service account. So I don't think wee need to put the event ID in the type field.

https://github.com/onflow/flow-go/blob/6990e52c7deef94844003e321551c5741d69f18f/fvm/handler/event.go#L160-L175

Reply:
That's true, but I don't think we can how these events are emitted in general. Added this issue to capture the change required here - can do separately https://github.com/dapperlabs/flow-go/issues/5588